### PR TITLE
fix embedding docs of jl_atexit_hook

### PR DESCRIPTION
**Pull request by KristofferC**
It seems that this now takes a return value parameter.

### CODE REVIEW

**Comment by KristofferC** [comment]
See for example https://github.com/JuliaLang/julia/blob/95b7080c13b4ca3c5390e4a3313a5a2af76da9da/examples/embedding.c

**Comment by yuyichao** [comment]
LGTM.

**Comment by tkelman** [comment]
also in `devdocs/eval.rst` and `devdocs/init.rst`

**Comment by KristofferC** [comment]
```
#. Just before exiting, :c:func:`main` calls `jl_atexit_hook() 
<https://github.com/JuliaLang/julia/blob/master/src/init.c>`_.
```

and

```
jl_atexit_hook()
----------------

:c:func:`main` calls `jl_atexit_hook()
<https://github.com/JuliaLang/julia/blob/master/src/init.c>`_.  This
calls _atexit for each module,
```

what would be good way to phrase this then?

**Comment by mlubin** [comment]
Bump

**Comment by KristofferC** [comment]
I just need to know what would be a good change in the two other places where `jl_atexit_hook` is mentioned. Tell me and I update it.

**Comment by KristofferC** [comment]
For first

```
#. Just before exiting, :c:func:`main` calls `jl_atexit_hook(int ret) 
```

and second:

```
jl_atexit_hook(int ret)
----------------

:c:func:`main` calls `jl_atexit_hook(ret)
<https://github.com/JuliaLang/julia/blob/master/src/init.c>`_.  This
calls _atexit for each module,
```
?

File: doc/manual/embedding.rst [hunk 1]
```diff
@@ -29,8 +29,9 @@ We start with a simple C program that initializes Julia and calls some Julia cod
            julia time to cleanup pending write requests
            and run all finalizers
       */
-      jl_atexit_hook();
-      return 0;
+      int ret = 0;
+      jl_atexit_hook(ret);
```
Comment (mlubin) [inline doc/manual/embedding.rst]: This seems misleading, ret is not a return value of ``jl_atexit_hook``. It's the exit code.  Why not just say:
```
jl_atexit_hook(0);
return 0;
```

Comment (KristofferC) [inline doc/manual/embedding.rst]: I just took it verbatim from https://github.com/JuliaLang/julia/blob/master/examples/embedding.c

Comment (mlubin) [inline doc/manual/embedding.rst]: Doesn't mean that's a good example...

Comment (KristofferC) [inline doc/manual/embedding.rst]: Also https://github.com/JuliaLang/julia/blob/1b51008291074e490e14ef6306fe23a7d9094b1a/ui/repl.c#L627

Comment (mlubin) [inline doc/manual/embedding.rst]: In that context it makes sense. I'd argue that it doesn't here.

Comment (mlubin) [inline doc/manual/embedding.rst]: At least rename ``ret`` to ``exitcode``, but it really seems unnecessary. You don't end an example ``main()`` function in C with:
```
int exitcode = 0;
return exitcode;
```

Comment (KristofferC) [inline doc/manual/embedding.rst]: Updated, I'll squash everything when the rest is good.

**Comment by mlubin** [comment]
Thanks. For the dev docs, just say that the argument is the exit code?

**Comment by KristofferC** [comment]
ah crap, i should have [c.i] skipped the last commits, oh well...

**Comment by KristofferC** [comment]
Maybe the devdocs should be left alone? There are a bunch of functions there that takes arguments but are just described as `foo()`.

**Comment by mlubin** [comment]
Doesn't bother me

**Comment by tkelman** [comment]
I think this version's okay. Please squash, without ci skip.

**Comment by KristofferC** [comment]
Done

**Comment by KristofferC** [comment]
Good to go?

**Comment by tkelman** [comment]
Yep, thanks!

**Pull request merged**